### PR TITLE
cloud: deploy: Add guide section

### PR DESCRIPTION
**Pull request by joaoaneto**

### CODE REVIEW

File: doc/cloud/cloud-deploy.rst [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+======
+
+This section will help you to deploy the KNoT Cloud.
+
+----------------------------------------------------------------
+
+Installation and usage
+----------------------
+
+Stacks for development and production environments are provided in order to assist the user's needs. The development stack must be used if one needs to modify any of the components of the stack. A command line tool will download the source for each component and plug it into the containers, which all have *hot reload* enabled. The production stack must be used in all other cases.
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: Please, use a maximum of 79 characters per line.

File: doc/cloud/cloud-deploy.rst [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+----------------------------------------------------------------
+
+Development only preparation
+----------------------------
+
+If you intend to use the development stack, a command line tool that downloads and configures the stack is required.
+
+Build and install CLI tool
+'''''''''''''''''''''''''''
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: Please, remove extra **'**

File: doc/cloud/cloud-deploy.rst [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+----------------------------------------------------------------
+
+Choose the stack
+----------------
+
+Development
+'''''''''''
+
+The development stack must be created using the tool described in the previous section. Choose a ``<path>`` where the stack should be created (defaults to current directory) and then run:
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: Please, use 
> (defaults to **the** current directory)

File: doc/cloud/cloud-deploy.rst [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+Initialize Swarm mode
+---------------------
+
+In your deployment machine, initialize Docker Swarm mode:
+
+   .. code-block:: bash
+
+        docker swarm init
+
+In case you are deploying to multiple nodes, all the nodes must be connected to the same swarm. In this case, run the command above in the machine that must be the swarm manager and run ``docker swarm join`` in all other nodes, as instructed by the execution of ``docker swarm init`` in the manager machine. Check the `Docker Swarm <https://docs.docker.com/engine/swarm/>`_ documentation for more on how to setup your cluster.
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: Please, use 
> Check the `Docker Swarm <https://docs.docker.com/engine/swarm/>`_ documentation for more on how to **set up** your cluster.

File: doc/cloud/cloud-deploy.rst [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+----------------------------------------------------------------
+
+Configure DNS
+-------------
+
+This stack uses `Traefik <https://traefik.io>`_ as a reverse proxy for the services and requires that you configure your DNS server to point, at least, the **www**, **ws**, **auth** and **bootstrap** subdomains to the machine where the stack is being deployed, so that it can route the requests to the appropriated service. It is possible to configure it to route by path or port, but instructions for that won't be provided here for brevity.
+
+If you don't have a domain or can't configure the main DNS server, you can configure a test domain in your machine before proceeding. Either setup a local DNS server, e.g. `bind9 <https://wiki.debian.org/Bind9>`_, or alternatively update your hosts file to include the following addresses:
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: Please, use
> Either **set up** a local DNS server

File: doc/cloud/cloud-deploy.rst [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+   .. code-block:: text
+
+        127.0.0.1	www
+        127.0.0.1	ws
+        127.0.0.1	auth
+        127.0.0.1	bootstrap
+
+On Windows, the hosts file is usually located under ``c:\Windows\System32\Drivers\etc\hosts``. On Unix systems, it is commonly found at ``/etc/hosts``. Regardless of you operating system, administrator or super user privileges will be required.
```
Comment (jcorde) [inline doc/cloud/cloud-deploy.rst]: > Regardless of **your** operating system, administrator or super user privileges will be required.

Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

File: doc/cloud/cloud-deploy.rst (hunk 5 above)
Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

File: doc/cloud/cloud-deploy.rst (hunk 4 above)
Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

File: doc/cloud/cloud-deploy.rst (hunk 3 above)
Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

File: doc/cloud/cloud-deploy.rst (hunk 2 above)
Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

File: doc/cloud/cloud-deploy.rst (hunk 1 above)
Comment (joaoaneto) [reply doc/cloud/cloud-deploy.rst]: Done.

**Pull request merged**